services/github/src/tc-yaml.js

```diff
@@ -199,9 +200,11 @@ class VersionOne extends TcYaml {
           prefix + payload.details['event.base.repo.branch'],
         ];
       }
-    } else if (payload.tasks_for === 'github-release') {
+    } else if (payload.tasks_for === GITHUB_TASKS_FOR.RELEASE) {
+      // Assume separate role if this is a pre-release action
+      const role = payload.action === GITHUB_RELEASE_ACTION.PRERELEASED ? 'pre-release' : 'release';
```

Comment:
Does Taskcluster process all the release events? Or just `preleased` and `released`. I'm wondering if we're scheduling a whole bunch of unnecessary graphs has Github releases are created..

Reply:
Follow-up question, if we do process all event types, maybe we should assume a separate role for each of them? I think you're trying to make this change as backwards compatible as possible. But since even this is backwards incompatible, I think I'd vote do it properly even it means a higher chance of something breaking.

If we do end up assuming a separate role for each action, maybe the format should be something like:
```
assume:repo:github.com/${ payload.organization }/${ payload.repository }:release:${action}
```

That way in the configuration we could grant scopes to `release:*` if we didn't want to bother doing each action individually. But then this raises another point that `pull-requests` also have an action, so maybe we should make the same change there :p.

I do think this change would be a great improvement (both for pull-requests and releases). I guess it boils down to how much time you want to spend on it. I don't think there's anything blocked on this feature atm (I initially filed the issue for VPN, but we have since worked around it).

Reply:
hmm.. that is a valid point. The webhook handler didn't filter out those events by action: https://github.com/taskcluster/taskcluster/blob/main/services/github/src/api.js#L319-L326 

Which means that all release events go through (or as many as github sends) and are published into `release` exchange. Later those are handled by the [job handler](https://github.com/taskcluster/taskcluster/blob/main/services/github/src/handlers/job.js)

So I would assume that all will end up creating new tasks. If we didn't notice any unwanted builds in the past it might be that github was not sending too many of those? I think you mentioned you saw 3 events for each publish event. Unless I'm missing something, I think it should create 3 graphs in such case.